ledger.go

```diff
@@ -380,56 +383,186 @@ func (l *Ledger) PullTransactions() {
 			select {
```

Comment:
This whole `select` section can be removed since the very first thing the top of the loop does is this